include/cufinufft/impl.h

```diff
@@ -235,23 +227,31 @@ int cufinufft_makeplan_impl(int type, int dim, int *nmodes, int iflag, int ntran
 
   if (type == 1 || type == 2) {
     CUFINUFFT_BIGINT nf1 = 1, nf2 = 1, nf3 = 1;
-    set_nf_type12(d_plan->ms, d_plan->opts, d_plan->spopts, &nf1,
-                  d_plan->opts.gpu_obinsizex);
-    if (dim > 1)
-      set_nf_type12(d_plan->mt, d_plan->opts, d_plan->spopts, &nf2,
-                    d_plan->opts.gpu_obinsizey);
-    if (dim > 2)
-      set_nf_type12(d_plan->mu, d_plan->opts, d_plan->spopts, &nf3,
-                    d_plan->opts.gpu_obinsizez);
-
+    if (d_plan->opts.gpu_spreadinterponly) {
+      // spread/interp grid is precisely the user "mode" sizes, no upsampling
+      nf1 = d_plan->ms;
```

Comment:
@chaithyagr ...and here is the corresponding GPU code.  At the risk of repetition, since the user allocates an N1*N2 output array, spreading could not write to any other size without segfault. Agreed?